### PR TITLE
Message forward fix

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Ensure that invitation are not displayed in the room list when forwarding a message.
Also do some rework and add a small UI test.

## Motivation and context

Less bug.

## Tests

- Get invited to a room A, but do not accept the invite
- Forward a message and observe that the room A is not displayed in the room list to select a room.
- In this room list, try to filter per name, and check that everything is working fine. 

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.forward_ForwardMessagesView_null_ForwardMessagesView-Day-2_2_null_3,NEXUS_5,1.0,en].png
Comment (jmartinesp) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.forward_ForwardMessagesView_null_ForwardMessagesView-Day-2_2_null_3,NEXUS_5,1.0,en].png:1]: Maybe we should try to improve this error message? Not a blocker, but it gives the impression that the feature is unfinished.

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/forward/ForwardMessagesStateProvider.kt [hunk 1]
```diff
@@ -17,34 +17,31 @@
...
             ),
             aForwardMessagesState(
-                forwardingSucceeded = persistentListOf(RoomId("!room2:domain")),
+                forwardAction = AsyncAction.Success(
+                    listOf(RoomId("!room2:domain")),
+                )
             ),
             aForwardMessagesState(
-                error = Throwable("error"),
+                forwardAction = AsyncAction.Failure(Throwable("error")),
```
Comment (jmartinesp) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/forward/ForwardMessagesStateProvider.kt:36]: Ah ok, I guess the UI is directly displaying the error message from the exception.

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/roomlist/RoomListFilter.kt [hunk 2]
```diff
@@ -20,6 +20,7 @@ sealed interface RoomListFilter {
     companion object {
         /**
          * Create a filter that matches all the given filters.
+         * If not filters are provided, all the room will match.
```
Comment (jmartinesp) [inline libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/roomlist/RoomListFilter.kt:23]:
```suggestion
         * If no filters are provided all the rooms will match.
```

File: libraries/roomselect/impl/src/main/kotlin/io/element/android/libraries/roomselect/impl/RoomSelectSearchDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+private const val PAGE_SIZE = 30
+
+class RoomSelectSearchDataSource @Inject constructor(
```
Comment (jmartinesp) [inline libraries/roomselect/impl/src/main/kotlin/io/element/android/libraries/roomselect/impl/RoomSelectSearchDataSource.kt:37]: Maybe add docs explaining it'll return only joined rooms?

**Review by jmartinesp — APPROVED** [review]
LGTM, just some minor comments.

File: libraries/roomselect/impl/src/main/kotlin/io/element/android/libraries/roomselect/impl/RoomSelectSearchDataSource.kt (hunk 3 above)
Comment (bmarty) [reply libraries/roomselect/impl/src/main/kotlin/io/element/android/libraries/roomselect/impl/RoomSelectSearchDataSource.kt:41]: Yes, good idea: c37d9ca6a46d6361582fb55b49a6b54e6f4d9afb

**Pull request merged**